scripts/test_adv_IM.py

```diff
@@ -14,23 +24,81 @@
+        help="list of component that ran",
+    )
+
     args = parser.parse_args()
 
+    return args
+
+
+def main(sim_dir, adv_im_model, components, exit=True):
+
     failed = []
-    for model in args.adv_im_model:
-        csv_path = os.path.join(args.sim_dir, "IM_calc/{}.csv".format(model))
+
+    for model in adv_im_model:
+        # check if csv exist
+        csv_path = os.path.join(sim_dir, "IM_calc/{}.csv".format(model))
+        im_dir = get_im_calc_dir(sim_dir)
         # check file exist first to prevent crash
         try:
             df = pd.read_csv(csv_path)
         except FileNotFoundError:
             failed.append(model)
             print("csv for {} does not exist".format(model))
         else:
+            # check for null/nan
             if df.isnull().values.any():
                 failed.append(model)
+
+            # if csv exist:
+            # check station count, if match, skip rest of checks
+            csv_station_count = len(df.station.unique())
+            # glob for station folders
+            list_folders = [x for x in os.listdir(im_dir) if os.path.isdir(x)]
+            if csv_station_count == len(list_folders):
+                # station folder count matches csv.station.count
+                continue
+            else:
+                for station_dir in list_folders:
+                    # check if folder is a station run folder
+                    station_model_dir = os.path.join(station_dir, model)
+                    station_name = os.path.basename(station_dir)
+                    if not os.path.isdir(station_model_dir) or (
+                        station_name in df.station.unique()
+                    ):
+                        # does not match naming for a station run, possibly created by something else, skipping
+                        continue
+                    else:
+                        result = True
+                        for comp in components:
+                            component_outdir = os.path.join(station_model_dir, comp)
+                            result = (
+                                check_status(component_outdir, check_fail=True)
+                                and result
+                            )
+                        # check of "Failed"
+                        # no "Failed" in all Analysis*.txt == crashed
+                        # "Failed" in all Analysis*.txt == failed to converge (normal), continue
+                        if result != True:
+                            # logs shows not all analysis failed to converge, but station failed to create csv
+                            print(f"{sim_dir}: {station_name} failed to run on {model}")
+                            failed.append(model)
     if len(failed) != 0:
-        print("some csv has empty data. models: {}".format(failed))
-        sys.exit(1)
+        print("some runs have failed,. models: {}".format(failed))
+        if exit:
+            sys.exit(1)
```

Comment:
Is there a case where we want a failing test to have a return code of 0?
Should just be able to make it always return 1
As there is no --exit argument, this would help any automated testing too

Reply:
the idea here is when you run the script directly, it behaves like a one-time scripts: show you the result and a related return code.
But if you are importing it, you can choose to not exit (otherwise it will also break the parent script that calls it)
So it will be handy if you have another wrapper that import this function and needs to run it multiple times.

Reply:
Is there currently any scripts importing it? If there isn't, I'm on the side of that this is introducing complexity.

Reply:
Ah, I forgot about the print statements (going to a slurm out file) having all the details from within a run

Reply:
@jpa198 / @shot0829 Unresolving because I'd still like an answer to my question.

Reply:
@jasonmotha yes, the current script i used to wrap around it is bash, thats why the exit code is required for it to catch the result. 
but as i mentioned, if there's anyone decided to use the main function within their code, the unconditioned exit is going to cause trouble.
I personally think its a good idea too have a option to turn off potential problematic features.

Reply:
i think what i can do is to make it sys.exit by default, but return code when desired.

Reply:
such as 
sys.exit(1) if exit else return 1